docs/first_semantic_game/02.simple_chat_npcs.rst

```diff
@@ -1,163 +1,395 @@
+``ChatPlayer`` for player characters.
+
+Right-click the filesystem dock and select "New Script...". **Make sure the language 
+is set to C#** and it inherits from ``Area2D``.
+
+Save this script as ``ChatEntity.cs`` in the project's root directory.
+
+Exported Properties
+^^^^^^^^^^^^^^^^^^^
+
+We’ll add two properties: ``ChatName`` and ``ChatDescr``. ``ChatName`` stores the 
+entity's name and will be visible to any nearby entities, while ``ChatDescr`` 
+holds a description of the entity. Both properties are exported for easy 
+modification in the Godot inspector.
+
+Put the following inside the ``ChatEntity`` class.
 
 .. code-block:: csharp
 
-   public partial class Player : CharacterBody2D
-   {
-       ...
+   [Export]
+   public string ChatName = "";
 
-       // Called when the node enters the scene tree for the first time.
-       public override void _Ready()
-       {
-           ...
-       }
+   [Export(PropertyHint.MultilineText)]
+   public string ChatDescr = "";
 
-       // Called once per physics tick
-       public override void _PhysicsProcess(double delta)
-       {
-           ...
-       }
-   }
+After saving the script, attach it to the ``NPC`` node in the main scene.
+Inspect the properties of the NPC, you should see ``ChatName`` and ``ChatDescr``.
+
+Next, we will add methods for handling messages: ``SendMsg`` and ``ReceiveMsg``.
 
+The ``SendMsg`` method is called when the entity wants to send a message to another
+entity. It prints the message to the console and then emits a ``MsgSent`` signal
+with the message as an argument. 
 
-Our first ChatAI class
-----------------------
+``ReceiveMsg`` is called when the entity receives a message from another entity.
+We expect this method to be overridden by a child class because the behavior of
+receiving a message will be different for different types of entities.
 
-The `ChatAI` class to be used in the demo is responsible for setting up and
-managing the conversations between NPCs, it uses OpenAI's GPT model via Semantic
-Kernel to handle the interactions. 
+.. code-block:: csharp
+
+    [Signal]
+    public delegate void MsgSentEventHandler(string msg);
 
-In this class, the following methods are used:
+    public virtual void SendMsg(string msg)
+    {
+        GD.Print(ChatName + ": " + msg);
+        CallDeferred("emit_signal", "MsgSent", msg);
+    }
+
+    public virtual void ReceiveMsg(string msg)
+    {
+        // This function will be overridden later by a child class
+    }
 
-- `GetApiKey` for retrieving the OpenAI API key from .NET secrets storage
-- A method to initialize GPT model via Semantic Kernel
-- A method to receive message and respond asynchronously
+Finally, we need to add a field to store a reference to our current conversation
+partner. This will be used later to send messages to the other entity.
+This ``_inConvoWith`` field is declared as ``internal`` so that it can be accessed
+by the main game class.

```

Comment:
I accidentally skipped over this the first time through. Maybe add a code block just for the `_inConvoWith` field declaration so that this part is harder to miss?

Reply:
```suggestion
Finally, we need to add a field to store a reference to our current conversation
partner. This will be used later to send messages to the other entity.
This ``_inConvoWith`` field is declared as ``internal`` so that it can be accessed
by the main game class.

.. code-block:: csharp

    internal ChatEntity _inConvoWith;

```